### PR TITLE
Create ProductPageDataProviderV2

### PR DESCRIPTION
**Pull request by gustaveen**
## Describe your changes
Since the new ProductPage won't have a price form, remove price template from the provider. 

## Justify why they are needed
Feed product page data to the new product page

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by gustaveen** [comment]
* **#4647** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4647?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gustaveen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/components/ProductPage/ProductPageDataProviderV2.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+'use client'
+import { atom, useAtomValue } from 'jotai'
+import { useHydrateAtoms } from 'jotai/utils'
+import type { ReactNode } from 'react'
+import { type ProductPageData } from '@/components/ProductPage/getProductPageData'
+
+const productPageDataAtomV2 = atom<ProductPageData | null>(null)
```
Comment (alebedev) [inline apps/store/src/components/ProductPage/ProductPageDataProviderV2.tsx:7]: Why do we want new atom, is shape of data different from the old one?

Comment (alebedev) [reply apps/store/src/components/ProductPage/ProductPageDataProviderV2.tsx:7]: Also, if we need new atom, it should be atomFamily + React context for the key, similar to recently updated `ProductDataProvider`. This setup allows to not use  `dangerouslyForceHydrate` and makes navigation both cheaper and more stable

Comment (gustaveen) [reply apps/store/src/components/ProductPage/ProductPageDataProviderV2.tsx:7]: Ah right you are, the data is the same format. Should we reuse it between the 2 versions then you mean? @alebedev

Comment (alebedev) [reply apps/store/src/components/ProductPage/ProductPageDataProviderV2.tsx:7]: Yes, we can just use `productPageDataAtom` everywhere

File: apps/store/src/components/ProductPage/ProductPageDataProviderV2.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  children: ReactNode
+  productPageData: ProductPageData
+}
+
+export function ProductPageDataProviderV2({ children, productPageData }: Props) {
+  useSyncProductPageData(productPageData)
+  return children
+}
+
+const useSyncProductPageData = (productPageData: ProductPageData) => {
```
Comment (alebedev) [inline apps/store/src/components/ProductPage/ProductPageDataProviderV2.tsx:18]: If atom is the same, you can now reuse hooks from original `ProductPageDataProvider` - only provider itself should be different

**Review by alebedev — APPROVED** [review]

File: apps/store/src/components/ProductPage/ProductPageDataProviderV2.tsx (hunk 2 above)
Comment (gustaveen) [reply apps/store/src/components/ProductPage/ProductPageDataProviderV2.tsx:18]: Ah but of course! 👌

**Comment by gustaveen** [comment]
### Merge activity

* **Aug 28, 7:56 AM EDT**: @gustaveen merged this pull request with [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4647).

**Pull request merged**